ingestion/src/main.rs

```diff
@@ -66,25 +67,41 @@ async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
         }
     });
 
-    // Spawn kvkafka reader
-    #[cfg(feature = "kafka_prod")]
-    {
-        let kafka_group = args[1].to_string();
-        println!("Spawing kvkafka reader...");
-        tokio::spawn(lard_ingestion::kvkafka::read_and_insert(
-            db_pool.clone(),
-            kafka_group,
-        ));
-    }
+    // set up cancellation token and signal catcher for graceful shutdown
+    let cancel_token = CancellationToken::new();
+    tokio::spawn(util::signal_catcher(cancel_token.clone()));
 
     // Set up and run our server + database
     println!("Ingestion server started!");
-    lard_ingestion::run(
+    let ingestor = tokio::spawn(lard_ingestion::run(
         db_pool,
         PARAMCONV,
         permit_tables,
         rove_connector,
         qc_pipelines,
-    )
-    .await
+        cancel_token,
+    ));
+
+    #[cfg(feature = "kafka_prod")]
+    // Spawn kvkafka reader
+    {
+        let kafka_group = args[1].to_string();
+        println!("Spawning kvkafka reader...");
+        let kvkafka_reader = tokio::spawn(lard_ingestion::kvkafka::read_and_insert(
+            db_pool.clone(),
+            kafka_group,
+            cancel_token.clone(),
+        ));
```

Comment:
Need to switch around the `clone()` calls (for `db_pool` and `cancel_token`). I guess we are not building with the `kafka_prod` feature in the CI pipeline (maybe we should), so it compiles fine :sweat:

Reply:
Done!